test/php/gutenberg-coding-standards/Gutenberg/Sniffs/Commenting/FunctionCommentSinceTagSniff.php

```diff
@@ -48,90 +91,444 @@ public function process( File $phpcsFile, $stackPtr ) {
+			return;
+		}
+
+		if ( 'T_VARIABLE' === $token['type'] && Scopes::isOOProperty( $phpcsFile, $stackPtr ) ) {
+			$this->process_property_token( $phpcsFile, $stackPtr );
+		}
+	}
+
+	/**
+	 * Processes a token representing a function call that invokes a WordPress hook,
+	 * checking for a missing `@since` tag in its docblock.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position of the hook token in the stack.
+	 */
+	protected function process_hook( File $phpcs_file, $stack_pointer ) {
+		$tokens = $phpcs_file->getTokens();
+
+		// The content of the current token.
+		$hook_function = $tokens[ $stack_pointer ]['content'];
+
+		$hook_invocation_functions = array(
+			'do_action',
+			'do_action_ref_array',
+			'apply_filters',
+			'apply_filters_ref_array',
 		);
 
-		foreach ( $wrapping_tokens_to_check as $wrapping_token_to_check ) {
-			if ( false !== $phpcsFile->getCondition( $stackPtr, $wrapping_token_to_check, false ) ) {
-				// This sniff only processes functions, not class methods.
+		// Check if the current token content is one of the filter functions.
+		if ( ! in_array( $hook_function, $hook_invocation_functions, true ) ) {
+			// Not a hook.
+			return;
+		}
+
+		$missing_since_tag_error_message = sprintf(
+			'@since tag is missing for the "%s()" hook function.',
+			$hook_function
+		);
+
+		$violation_code = 'MissingHookSinceTag';
+
+		$docblock = static::find_hook_docblock( $phpcs_file, $stack_pointer );
+		if ( false === $docblock ) {
+			$phpcs_file->addError( $missing_since_tag_error_message, $stack_pointer, $violation_code );
+			return;
+		}
+
+		list( $doc_block_start_token, $doc_block_end_token ) = $docblock;
+
+		$version_token = static::parse_since_tags( $phpcs_file, $doc_block_start_token, $doc_block_end_token );
+		if ( false === $version_token ) {
+			$docblock_content = GetTokensAsString::compact( $phpcs_file, $doc_block_start_token, $doc_block_end_token, false );
+			if ( false !== stripos( $docblock_content, 'This filter is documented in ' ) ) {
+				// The hook is documented elsewhere.
 				return;
 			}
+
+			$phpcs_file->addError( $missing_since_tag_error_message, $stack_pointer, $violation_code );
+			return;
+		}
+
+		$version_tags = static::parse_since_tags( $phpcs_file, $doc_block_start_token, $doc_block_end_token );
+		if ( empty( $version_tags ) ) {
+			$docblock_content = GetTokensAsString::compact( $phpcs_file, $doc_block_start_token, $doc_block_end_token, false );
+			if ( false === stripos( $docblock_content, 'This filter is documented in ' ) ) {
+				$phpcs_file->addError( $missing_since_tag_error_message, $stack_pointer, $violation_code );
+			}
+
+			// The hook is documented elsewhere.
+			return;
+		}
+
+		foreach ( $version_tags as $since_tag_token => $version_value_token ) {
+			if ( null === $version_value_token ) {
+				$phpcs_file->addError( $missing_since_tag_error_message, $since_tag_token, $violation_code );
```

Comment:
Maybe the error message here could be different as the `@since` tag is not missing and what is missing is its value?

Reply:
I agree.